ecl/data_source_ecl_vna_appliance_plan_v1.go

```diff
@@ -0,0 +1,223 @@
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"availability_zones": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"availability_zone": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"available": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+						},
+						"rank": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceVNAAppliancePlanV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	var opts appliance_plans.ListOpts
+
+	if v, ok := d.GetOk("id"); ok {
+		opts.ID = v.(string)
+	}
+	if v, ok := d.GetOk("name"); ok {
+		opts.Name = v.(string)
+	}
+	if v, ok := d.GetOk("description"); ok {
+		opts.Description = v.(string)
+	}
+	if v, ok := d.GetOk("appliance_type"); ok {
+		opts.ApplianceType = v.(string)
+	}
+	if v, ok := d.GetOk("version"); ok {
+		opts.Version = v.(string)
+	}
+	if v, ok := d.GetOk("flavor"); ok {
+		opts.Flavor = v.(string)
+	}
+	if v, ok := d.GetOk("number_of_interfaces"); ok {
+		opts.NumberOfInterfaces = v.(int)
+	}
+	if v, ok := d.GetOk("enabled"); ok {
+		opts.Enabled = v.(bool)
+	}
+	if v, ok := d.GetOk("max_number_of_aap"); ok {
+		opts.MaxNumberOfAap = v.(int)
+	}
+	if v, ok := d.GetOk("availability_zone"); ok {
+		opts.AvailabilityZone = v.(string)
+	}
+	if v, ok := d.GetOk("availability_zone.available"); ok {
+		opts.AvailabilityZoneAvailable = v.(bool)
+	}
+
+	vnaClient, err := config.vnaV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating ECL virtual network appliance client: %s", err)
+	}
+
+	allPlans, err := getVirtualNetworkAppliancePlans(vnaClient, opts)
+	if err != nil {
+		return fmt.Errorf("error getting Load Balancer Plans: %w", err)
+	}
+
+	if len(allPlans) > 1 {
+		return fmt.Errorf("specified Virtual Network Appliance Plan query returned more than one result")
+	}
+
+	if len(allPlans) == 0 {
+		return fmt.Errorf("specified Virtual Network Appliance Plan query returned no results")
+	}
+
+	plan := allPlans[0]
+
+	log.Printf("[DEBUG] Retrieved Virtual Network Appliance Plan %s: %+v", plan.ID, plan)
+
+	d.SetId(plan.ID)
+	d.Set("id", plan.ID)
+	d.Set("name", plan.Name)
+	d.Set("description", plan.Description)
+	d.Set("appliance_type", plan.ApplianceType)
+	d.Set("version", plan.Version)
+	d.Set("flavor", plan.Flavor)
+	d.Set("number_of_interfaces", plan.NumberOfInterfaces)
+	d.Set("enabled", plan.Enabled)
+	d.Set("max_number_of_aap", plan.MaxNumberOfAap)
+
+	licenses := make([]map[string]string, len(plan.Licenses))
+	for _, l := range plan.Licenses {
+		license := map[string]string{"license_type": l.LicenseType}
+		licenses = append(licenses, license)
+	}
```

Comment:
```suggestion
	licenses := make([]map[string]string, len(plan.Licenses))
	for i, l := range plan.Licenses {
		license := map[string]string{"license_type": l.LicenseType}
		licenses[i] = license
	}
```
The list length is specified at the time of initialization, the length will be doubled when append.